lib/gateway/pipelines.js

```diff
@@ -190,12 +191,16 @@ function validateGatewayConfig (config) {
   if (!gatewayConfig.pipelines) {
     if (gatewayConfig.pipeline) {
       gatewayConfig.pipelines = Array.isArray(gatewayConfig.pipeline) ? gatewayConfig.pipeline : [ gatewayConfig.pipeline ];
-    } else throw new ConfigurationError('No pipelines found');
+    } else {
+      gatewayConfig.pipeline = [];
+    }
   }
   if (!gatewayConfig.apiEndpoints) {
     if (gatewayConfig.apiEndpoint) {
       gatewayConfig.apiEndpoints = Array.isArray(gatewayConfig.apiEndpoint) ? gatewayConfig.apiEndpoint : [ gatewayConfig.apiEndpoint ];
-    } else throw new ConfigurationError('No apiEndpoints found');
+    } else {
+      gatewayConfig.apiEndpoints = [];
```

Comment:
same suggestion as for pipelines
gatewayConfig.apiEndpoints = gatewayConfig.apiEndpoints || gatewayConfig.apiEndpoint || {} ?